src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -119,7 +119,11 @@ public HttpConnectionPool(HttpConnectionPoolManager poolManager, HttpConnectionK
             }
 
             _http2Enabled = _poolManager.Settings._maxHttpVersion >= HttpVersion.Version20;
-            _http3Enabled = _poolManager.Settings._maxHttpVersion >= HttpVersion.Version30 && (_poolManager.Settings._quicImplementationProvider ?? QuicImplementationProviders.Default).IsSupported;
+            // TODO: Replace with Platform-Guard Assertion Annotations once https://github.com/dotnet/runtime/issues/44922 is finished
+            if (OperatingSystem.IsLinux() || OperatingSystem.IsWindows() || OperatingSystem.IsMacOS())
```

Comment:
Hm, OSX should be accounted as MacOS but somehow that logic is left out of the analyzer, i will fix that soon.
Could you use macOS for the attribute as a workaround `<SupportedOSPlatforms>Windows;Linux;MacOS</SupportedOSPlatforms>`?

Reply:
It still didn't help, when I run build locally I still get tons of errors for `OSX` and also `FreeBSD`, `iOS`, `Android`, etc. See https://gist.github.com/ManickaP/7d2ca0d2bb18d27c12e79a16795ee9a9

---

For instance this: https://gist.github.com/ManickaP/7d2ca0d2bb18d27c12e79a16795ee9a9#file-buils_errors-L50:
```
/home/manicka/Repositories/runtime.2/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs(31,87): error CA1416: This call site is reachable on: 'OSX'. 'QuicStream' is only supported on: 'MacOS', 'Windows', 'Linux'. [/home/manicka/Repositories/runtime.2/src/libraries/System.Net.Http/src/System.Net.Http.csproj]
```
Points to a field of `Http3Connection`:
https://github.com/dotnet/runtime/blob/4831c4c8bd24f8b63ea674719e3dcfa355ff3110/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs#L31
which is never initialized.

---

Another example, more direct,  is https://gist.github.com/ManickaP/7d2ca0d2bb18d27c12e79a16795ee9a9#file-buils_errors-L94:
```
/home/manicka/Repositories/runtime.2/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs(796,115): error CA1416: This call site is reachable on: 'OSX'. 'QuicImplementationProviders.Default' is only supported on: 'MacOS', 'Windows', 'Linux'. [/home/manicka/Repositories/runtime.2/src/libraries/System.Net.Http/src/System.Net.Http.csproj]
```
Code:
https://github.com/dotnet/runtime/blob/4831c4c8bd24f8b63ea674719e3dcfa355ff3110/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L796
residing in a method `GetHttp3ConnectionAsync` that's called from a single place:
https://github.com/dotnet/runtime/blob/b737eca18e330028e1ddbc3ae6018157c8d70064/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L383-L383
which has the guards in place.

Reply:
The analyzer does not follow any method call flow. You have to annotate every method body even if it's not reachable. See my response earlier https://github.com/dotnet/runtime/pull/49261#issuecomment-791934484

Reply:
Should be addressed now.
I sprinkled some `[SupportedOSPlatform("...")]` and it seemed to help.

Reply:
JFYI: `OSX should be accounted as MacOS` scenario is not fixed in the analyzer, if you like to use `OSX` for attributes you can do that after updating the analyzer version into the latest build `6.0.0-preview4.21212.4`.